docs/tutorials/optimizers_cyclicallearningrate.ipynb

```diff
@@ -0,0 +1,450 @@
+        "\n",
+        "Below, you visualize how the `clr` schedule looks like. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "b_WRGfDx4Wwc"
+      },
+      "outputs": [],
+      "source": [
+        "step = np.arange(0, EPOCHS * steps_per_epoch)\n",
+        "lr = clr(step)\n",
+        "plt.plot(step, lr)\n",
+        "plt.xlabel(\"Steps\")\n",
+        "plt.ylabel(\"Learning Rate\")\n",
+        "plt.show()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bBlKaAqNjHP1"
+      },
+      "source": [
+        "In order to better visualize the effect of CLR, you can plot the schedule with an increased number of steps. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Gjhoyk-Li368"
+      },
+      "outputs": [],
+      "source": [
+        "step = np.arange(0, 100 * steps_per_epoch)\n",
+        "lr = clr(step)\n",
+        "plt.plot(step, lr)\n",
+        "plt.xlabel(\"Steps\")\n",
+        "plt.ylabel(\"Learning Rate\")\n",
+        "plt.show()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ObYcy5NRkF4V"
+      },
+      "source": [
+        "The function you are using in this tutorial is referred to as the `triangular2` method in the CLR paper. There are other two functions there were explored namely `triangular` and `exp` (short for exponential). "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "-OV_8QVIe5m_"
+      },
+      "source": [
+        "## Train a model with CLR"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "zRSglElvy_fF"
+      },
+      "outputs": [],
+      "source": [
+        "clr_model = tf.keras.models.load_model(\"initial_model\")\n",
+        "clr_history = train_model(clr_model, optimizer=optimizer)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "8rhTLQJdnGfP"
+      },
+      "source": [
+        "As expected the loss starts higher than the usual and then it stabilizes as the cycles progress. You can confirm this visually with the plots below. "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LyHEgnv6e8lX"
+      },
+      "source": [
+        "## Visualize losses"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "wg0JjLwH2RTl"
+      },
+      "outputs": [],
+      "source": [
+        "(fig, ax) = plt.subplots(1, 2)\n",
```

Comment:
It would be good for this plot of the two subplots had the same `ylim`. Also the lower y-lim should be 0.